recipes/cubicinterpolation/all/conanfile.py

```diff
@@ -23,91 +26,84 @@ class CubicInterpolationConan(ConanFile):
-        return "source_subfolder"
-
     @property
     def _is_msvc(self):
         return str(self.settings.compiler) in ["Visual Studio", "msvc"]
 
+    def export_sources(self):
+        #self.copy("CMakeLists.txt")
+        export_conandata_patches(self)
+
     def config_options(self):
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         # TODO: update boost dependency as soon as issue #11207 is fixed
         self.requires("boost/1.75.0")
         self.requires("eigen/3.3.9")
 
-    @property
-    def _minimum_compilers_version(self):
-        return {
-            "Visual Studio": "16",
-            "gcc": "5",
-            "clang": "5",
-            "apple-clang": "5.1",
-        }
-
     @property
     def _required_boost_components(self):
         return ["filesystem", "math", "serialization"]
 
     def validate(self):
-        miss_boost_required_comp = any(getattr(self.options["boost"], "without_{}".format(boost_comp), True) for boost_comp in self._required_boost_components)
-        if self.options["boost"].header_only or miss_boost_required_comp:
-            raise ConanInvalidConfiguration("{0} requires non header-only boost with these components: {1}".format(self.name, ", ".join(self._required_boost_components)))
 
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "14")
-
-        minimum_version = self._minimum_compilers_version.get(
-            str(self.settings.compiler), False
-        )
-        if not minimum_version:
-            self.output.warn(
-                "CubicInterpolation requires C++14. Your compiler is unknown. Assuming it supports C++14."
-            )
-        elif tools.Version(self.settings.compiler.version) < minimum_version:
+        miss_boost_required_comp = any(getattr(self.dependencies["boost"].options, f"without_{boost_comp}", True) for boost_comp in self._required_boost_components)
+        if self.dependencies["boost"].options.header_only or miss_boost_required_comp:
             raise ConanInvalidConfiguration(
-                "CubicInterpolation requires C++14, which your compiler does not support."
+                f"{self.ref} requires non header-only boost with these components: "
+                f"{', '.join(self._required_boost_components)}",
             )
 
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, "14")
+
+        if str(self.settings.compiler) == "Visual Studio" and Version(self.settings.compiler.version) < "16":    
```

Comment:
https://docs.conan.io/2/reference/tools/microsoft/helpers.html?highlight=check%20min%20vs#conan.tools.microsoft.visual.check_min_vs

is raise = False would be better

Reply:
`check_min_vs` with `raise_invalid=False` has only been available since `1.57.0` (see #12880).

Should I just replace the block with a simple `check_min_vs(self, 192)`? Otherwise I would have to change the required_conan_version to `>=1.57.0`.

Reply:
ConanCenter recipes are always the latest (minus a version) Consumers are expected to be on 1.59 (sooner the later it will become 2.0) so it's absolutely allowed to increase the min version :)